erpnext/selling/doctype/sales_order/sales_order.py

```diff
@@ -181,6 +185,9 @@ def on_submit(self):
 		self.update_prevdoc_status('submit')
 
 		self.update_blanket_order()
+		if self.coupon_code:
+			from erpnext.accounts.doctype.coupon_code.coupon_code import update_coupon_code_count
+			update_coupon_code_count(self.coupon_code)
```

Comment:
on_cancel ????

Reply:
Good Catch! Now on cancel the count of used coupon is reduced. thanks